### ISSUE TITLE
`prefer-top-level-await` falsely flagging zod `.catch` calls as promises

### ISSUE THREAD

**Issue by briavicenti** [sindresorhus/eslint-plugin-unicorn#2149]
In our Typescript app, we frequently use the zod `.catch` helper (see [here](https://zod.dev/?id=catch)) to add fallback behavior to schemas used to validate API data. These schema setups are being incorrectly flagged by the `prefer-top-level-await` rule as promises that should be given a top-level await instead.

Example:
```
import z from "zod";

const someSchema = z.string().catch("");
```

This rule just appears to error based on name, so this issue exists for any method called `catch`. These also both error:
```
const objectWithCatch = {
  catch: () => undefined,
};

objectWithCatch.catch(); // errors

const getCatchMethod = () => ({
  catch: () => undefined,
});

getCatchMethod().catch();
```

It would be great to pull typing information into this rule to make sure that the `catch` in question actually pertains to a promise.

**Comment by sindresorhus** [comment]
> z.string().catch("");

This case we can fix without type info by checking by ignoring it if the root is called `z`.

**Comment by osdiab** [comment]
would that work even if I did this?

```
const schema = z.string();
const schemaWithFallback = schema.catch("fallback");
```

since that's the situation where it's a false positive for me.

**Comment by kachkaev** [comment]
> would that work even if I did this?

I guess we can check `z ××××× .catch()` as well as `schema ××××× .catch()` / `fooSchema ××××× .catch()`. Should be good enough.
